subsys/bluetooth/mesh/access.c

```diff
@@ -1892,7 +1892,16 @@ static int comp_set(const char *name, size_t len_rd, settings_read_cb read_cb,
 
 	return 0;
 }
-BT_MESH_SETTINGS_DEFINE(comp, "cmp", comp_set);
+BT_MESH_SETTINGS_DEFINE(comp, "cmp", comp_p0_set);
+
+static int comp_p1_set(const char *name, size_t len_rd, settings_read_cb read_cb,
+		    void *cb_arg)
+{
+	/* Identical behavior as for page 0 */
+	comp_p0_set(name, len_rd, read_cb, cb_arg);
+	return 0;
+}
+BT_MESH_SETTINGS_DEFINE(comp1, "cmp1", comp_p1_set);
```

Comment:
If the behavior is identical, the set handler is probably not needed?
```suggestion
BT_MESH_SETTINGS_DEFINE(comp1, "cmp1", comp_p0_set);
```

Reply:
I belive I had some issues with this:

I tried using the same CB for both, I.E. `BT_MESH_SETTINGS_DEFINE(comp1, "cmp1", comp_p0_set);` -> /zephyr/subsys/bluetooth/mesh/settings.h:30:20: error: redefinition of ‘pre_comp_p0_set’.

I also tried `BT_MESH_SETTINGS_DEFINE(comp1, "cmp1", NULL);` -> zephyr/subsys/bluetooth/mesh/access.c:1904:40: error: called object is not a function or function pointer.

So I was not able to find any other way to do this, any advice?

Reply:
```
static int comp_set(const char *name, size_t len_rd, settings_read_cb read_cb,
		    void *cb_arg)
{
	if (!atomic_test_bit(bt_mesh.flags, BT_MESH_INIT)) {
		return 0;
	}

	/* Only need to know that the entry exists. Will load the contents on
	 * demand.
	 */
	if (len_rd > 0) {
		atomic_set_bit(bt_mesh.flags, BT_MESH_COMP_DIRTY);
	}

	return 0;
}

#ifdef CONFIG_BT_SETTINGS
SETTINGS_STATIC_HANDLER_DEFINE(bt_mesh_comp, "bt/mesh/cmp", NULL, comp_set, NULL, NULL);
SETTINGS_STATIC_HANDLER_DEFINE(bt_mesh_comp1, "bt/mesh/cmp1", NULL, comp_set, NULL, NULL);

#else
/* Declaring non static settings handler helps avoid unnecessary ifdefs
 * as well as unused function warning. Since the declared handler structure is
 * unused, linker will discard it.
 */
const struct settings_handler settings_handler_bt_mesh_comp = {
	.h_set = comp_set,
};

#endif
```

This is one alterantive that is possible. `BT_MESH_SETTINGS_DEFINE` is not designed to handle reuse of functions, so alternatively we need to create a separate macro in `settings.h` to handle this particular case.

Reply:
To be honset im not sure if this bloat is worth it to avoid the redundant `comp1` set cb

Reply:
What if we store CDPs like this: `bt/mesh/cmp/0`, `bt/mesh/cmp/1`, ...? Then the old code will work as is, but will be just called as many times as number of pages that were stored.

Reply:
> What if we store CDPs like this: `bt/mesh/cmp/0`, `bt/mesh/cmp/1`, ...? Then the old code will work as is, but will be just called as many times as number of pages that were stored.

Will this be backwards compatible?

Reply:
It is part of mesh-1.1 features, it shouldn't have been used before.

Reply:
I mean, Alex mentioned earlier in this PR that changing how cmp0 is stored might cause issues if the new scheme is part of a DFU update. I just wanted to mention it since he won't be able to participate in the rest of this PR. But if you think that this is not an issue im fine with doing it like this 👍

Reply:
Replied in that conversation